### PR TITLE
Add CSV import audit CLI task, refs #13569

### PR DESCRIPTION
**Pull request by mcantelon**
Adds CLI task to check legacy ID values from a CSV import file and list any that haven't been imported yet (creating the list by checking keymap data).

### CODE REVIEW

File: lib/CsvImportAuditer.class.php [hunk 1]
```diff
@@ -0,0 +1,308 @@
...
+
+                continue;
+            }
+
+            ++$this->rowsAudited;
+            $this->log($this->progressUpdate($this->rowsAudited));
+        }
+
+        if (!empty($this->missingIds)) {
+            $this->log('');
```
Comment (djjuhasz) [inline lib/CsvImportAuditer.class.php]: Is logging an empty string here necessary?

File: lib/CsvImportAuditer.class.php [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+    {
+        // Write to error log (but not STDERR)
+        if (STDERR != $this->getErrorLogHandle()) {
+            fwrite($this->getErrorLogHandle(), $msg.PHP_EOL);
+        }
+    }
+
+    protected function getErrorLogHandle(): resource
+    {
+        if (null === $filename = $this->getOption('errorLog')) {
```
Comment (djjuhasz) [inline lib/CsvImportAuditer.class.php:274]: Looks like this is expecting a `$filename` argument for this method that doesn't exist?

File: lib/task/import/csvAuditImportTask.class.php [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+        $this->namespace = 'csv';
+        $this->name = 'audit-import';
+        $this->briefDescription = 'Audit CSV import.';
+        $this->detailedDescription = <<<'EOF'
+Audit CSV import by checking to make sure a keymap has been created for each
+row.
+EOF;
+    }
+
+    protected function getDbConnection()
```
Comment (djjuhasz) [inline lib/task/import/csvAuditImportTask.class.php]: It doesn't look like this method is called?

File: test/phpunit/CsvImportAuditerTest.php [hunk 4]
```diff
@@ -0,0 +1,368 @@
+<?php
+
+use org\bovigo\vfs\vfsStream;
+
+/**
+ * @internal
+ * @covers \CsvImportAuditer
+ */
+class CsvImportAuditerTest extends \PHPUnit\Framework\TestCase
```
Comment (djjuhasz) [inline test/phpunit/CsvImportAuditerTest.php:9]: YAY UNIT TESTS!!! 🥇 🎉 👍

**Review by djjuhasz — CHANGES_REQUESTED** [review]
Looks good @mcantelon, just a few minor change requests.  Love the unit tests! 😁

File: lib/CsvImportAuditer.class.php (hunk 1 above)
Comment (mcantelon) [reply lib/CsvImportAuditer.class.php]: I was hacking in a blank line but I'll see if I can do that but explicitly adding one instead.

File: lib/CsvImportAuditer.class.php (hunk 2 above)
Comment (mcantelon) [reply lib/CsvImportAuditer.class.php:274]: It's getting the value, if set, from the "errorLog" option, if set.

**Comment by mcantelon** [comment]
Thanks @djjuhasz ! I've updated the PR.

File: lib/CsvImportAuditer.class.php (hunk 2 above)
Comment (djjuhasz) [reply lib/CsvImportAuditer.class.php:274]: Oh right. :)

**Review by djjuhasz — APPROVED** [review]
Looks good @mcantelon! :shipit:

**Pull request merged**